enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,560 @@
+indicates that cloud provider is external and the operator encounters a platform which it is not aware of it will go into 'Degraded' state.
+
+During the first phases of the "External" platform type enablement, this operator should be just disabled.
+This might be done with changes within the [library-go](#library-go) and further dependency updates or, better,
+by adding a respective check within the operator itself.
+
+In the future, when the delivery mechanism for CCMs will be defined,
+the operator might be engaged for deploying a user-provided cloud controller manager,
+however this is a subject for the upcoming design work.
+
+#### Machine Api Operator
+
+Machine Api Operator is responsible for deploying and maintaining the set of machine-api related controllers, such as
+
+- machineset controller
+- nodelink controller
+- machine health check controller
+- machine controller
+
+From the list above, only the "machine controller" is cloud-provider dependent, however, for now, Machine Api Operator
+won't deploy anything if it encounters "None" or an unrecognized platform type. 
+
+In the future, "External" platform type, in conjunction with an enabled capability,
+would serve as a signal for Machine Api Operator to deploy only provider-agnostic
+controllers, which would leave room for the user to supplement only the machine controller and not to reverse engineer and
+replicate everything that MAO does.
+
+#### Cluster Storage Operator
+
+Cluster Storage Operator will go to no-op state if it encounters PlatformType "None" or an unknown PlatformType.
+
+At this point, nothing requires storage to be there during cluster installation, and storage (CSI) drivers might be supplemented
+later via OLM or some other way as day two operation. 
+
+No particular changes in regards to the "External" platform type introduction are expected there.
+
+#### Cloud Credential Operator
+
+Cloud Credential Operator is responsible for handling `CredentialsRequest` custom resource.
+`CredentialsRequest`s allow OpenShift components to request credentials for a particular cloud provider.
+
+On unsupported platforms, the operator goes into no-op mode, which technically is mostly an equivalent
+of the ["Manual"](https://docs.openshift.com/container-platform/4.11/authentication/managing_cloud_provider_credentials/cco-mode-manual.html)
+mode.
+
+Cloud Credential operator [uses](https://github.com/openshift/cloud-credential-operator/blob/master/pkg/operator/controller.go#L74) an "actuator pattern"
+and, in theory, in the future, it might be extended in a way to react to the new "External" platform type and allow users to supplement their own
+platform-specific credentials management logic.
+
+During initial enablement phases of "External" platform type, no specific actions will be needed there, since
+CCO [would](https://github.com/openshift/cloud-credential-operator/blob/master/pkg/operator/controller.go#L134) go
```

Comment:
nit: this link still points at the `default` branch, but it might be worth using `58c41771a87e613415a1fa16d299601a1c2f48c2` instead of `master` to pin that reference, so folks know what you were pointing at, even if that file moves or has lines shifted in the future.